azkaban-common/src/main/java/azkaban/server/AbstractMBeanRegistrableServer.java

```diff
@@ -0,0 +1,116 @@
+/*
+ * Copyright 2019 LinkedIn Corp.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+package azkaban.server;
+
+import java.lang.management.ManagementFactory;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.TreeMap;
+import javax.management.MBeanAttributeInfo;
+import javax.management.MBeanInfo;
+import javax.management.MBeanServer;
+import javax.management.ObjectName;
+import org.apache.log4j.Logger;
+
+
+/**
+ * Abstract class which can register MBeans
+ */
+public abstract class AbstractMBeanRegistrableServer {
+  private static final Logger logger = Logger.getLogger(AbstractMBeanRegistrableServer.class);
+
+  private List<ObjectName> registeredMBeans = new ArrayList<>();
+  private MBeanServer mbeanServer = null;
+
+  protected abstract void configureMBeanServer();
```

Comment:
Please add some comments for the functions.

Reply:
discuss with chen offline. I am gonna refactor this by a composition way. and, I will add more comments for sure.